x-pack/filebeat/input/cel/config_auth.go

```diff
@@ -341,7 +343,14 @@ func (o *oAuth2Config) validateOktaProvider() error {
 	}
 	// jwk_pem
 	if o.OktaJWKPEM != "" {
-		_, err := x509.ParsePKCS1PrivateKey([]byte(o.OktaJWKPEM))
+		blk, rest := pem.Decode([]byte(o.OktaJWKPEM))
+		if rest := bytes.TrimSpace(rest); len(rest) != 0 {
+			return fmt.Errorf("PEM text has trailing data: %s", rest)
```

Comment:
I think a check for `blk == nil` is warranted if it is going to check for extra data in `rest`.

And to be safe against logging key material, I would omit `rest` from the error.

Reply:
The data in `rest` is partially orthogonal to the nilness of `blk` because of the `bytes.TrimSpace` call, but I have guarded against a nil deref.